README.md

```diff
@@ -1,24 +1,161 @@
 
-## Objectives
+## Review
 
+You already know how to create a static request, which is where you create a page that doesn't take any parameters and simply renders a view, an example would be: ```localhost:3000/about```. For Rails to process this request, the ```route.rb``` file contains a route such as:
 
-1. Explain the differences between a static route and a dynamic route with regards to variables
-2. Write a dynamic route with the :variable syntax
-3. Access a route variable value from params
-4. Use a route variable value in params within a controller action (like in Post.find(params[:id])
-5. Use the convention of naming the primary key :id as a route variable
-6. Pass an instance variable to the view
-7. use a route variable to load a object and render that show page
+```ruby
+get 'about', to "static#about"
+```
 
-## Notes
+This is mapped to the ```static``` controller and ```about``` action, which renders the ```about.html.erb``` view template.
 
-The next restful action we want to build is the ability to show a post.
+## Dynamic Requests
 
-to do that we need to integrate route variables into our routes for a post /posts/1 can't be hardcoded.
+If you are comfortable with how static requests you will pick up on dynamic requests quickly. An example scenario for a dynamic request would be where a user navigates to a page in the application such as ```localhost:3000/posts/42``` and is displayed a post with the id of 42. Behind the scenes what is happening is below:
 
-how to draw a dynamic route, :variable syntax and how that is arbitrary but will correspond to params.
+1. The ```route.rb``` file takes in the request and processes it like normal, except this time it also parses the ```42``` parameter and passes it to the posts' controller.
 
-how to access the dynamic route variable in a controller action and param how to use that data in find Post
+2. From that point the controller parses the ```42``` parameter and runs a query on the Post model, storing the result in an instance variable.
 
-passing an instance variable to a view
+3. Lastly the controller passes the instance variable to the associated view which renders that specific post record details to the client.
+
+
+## Code Implementation
+
+In order to setup a dynamic request feature we will start by writing a test to verify that the page exists (you can ignore the FactoryGirl calls, I'm simply using it to create a new post so we have an object to work with):
+
+```ruby
+# spec/features/post_spec.rb
+
+require 'rails_helper'
+
+feature 'navigate to post pages' do
+  subject { FactoryGirl.create(:post) }
+
+  scenario 'on the show page' do
+    visit "/posts/#{subject.id}"
+    expect(page.status_code).to eq(200)
+  end
+
+end
+```
+
+Running ```bundle exec rspec``` gives us an expected error of: ```ActionController::RoutingError: No route matches [GET] "/posts/1"```. To correct this error let's draw a route that maps to a show action in the posts' controller:
```

Comment:
@jmburges on this I always run it because it's recommended to do it on servers since it protects against some weird edge cases errors where bundler has different gem versions on the server vs. the app itself, so it's just a matter of habit for me.